core/src/mender-client.c

```diff
@@ -1092,6 +1092,11 @@ mender_client_check_deployment(mender_api_deployment_data_t **deployment_data) {
     cJSON_AddStringToObject(mender_client_deployment_data, "artifact_name", deployment->artifact_name);
     cJSON_AddArrayToObject(mender_client_deployment_data, "types");
 
+    assert(NULL != mender_client_deployment_data);
```

Comment:
This is guaranteed by the `if` a couple lines above, right?

Reply:
Yep